lms/djangoapps/course_api/blocks/transformers/video_urls.py

```diff
@@ -46,7 +46,11 @@ def transform(self, usage_info, block_structure):
             )
             if not student_view_data:
                 return
-            encoded_videos = student_view_data['encoded_videos']
+
+            # encoded_videos key doesn't exist for web-only videos
+            encoded_videos = student_view_data.get('encoded_videos', False)
```

Comment:
Looks good to me, Just a few questions.

When this case is true, will the `student_view_data` will contain `only_on_web` flag? 
Are these two are coupled together?
Is there any other case where `student_view_data.get('encoded_videos', False)` can be `False` other than `only_on_web` flag?

What about `student_view_data.get(only_on_web) and student_view_data.get('encoded_videos')`?

Reply:
@awaisdar001 Yes, when the video is web-only, the student_view_data will only contain `only_on_web` key. The `student_view_data` for the video module has two return statements. One is at the very start, which checks that if the video is web-only, return that key only. If not, it gets all the required information for the video to be served to native clients([Source](https://github.com/edx/edx-platform/blob/master/common/lib/xmodule/xmodule/video_module/video_module.py#L1060-L1138)). 
Given the current code structure, the encoded_videos will be False only when the video is web-only. There are no other cases in the method that will make it False.
As for the last part, `student_view_data.get('encoded_videos')` will yield the same result(and I will cover that nit). I deliberately didn't do the check with `only_on_web`. Since our concern is with encoded_videos key, it would be much simpler to check for that key's existence and stop work if the key doesn't exist for the video.

Reply:
I understand that the current state of the method only has two `return` statements.  

My concern here is; if we are only dealing with `only_on_web` case, we should be explicit.
The current `if` condition has a tendency to swallow server errors and edge cases. Let's be explicit about the case we are handling.

Reply:
ok